### DISCUSSION TITLE
how to install c compiler for eclipse

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-cdt/cdt#543] [Q&A]
I installed mingw64 and downloaded eclipse for c/c++.  I added c:\msys64\ming64\bin to my path environment variable.  When I try to run c project hello.c I get the following error:

20:11:49 **** Rebuild of configuration Debug for project hello ****
Info: Configuration "Debug" uses tool-chain "MinGW GCC" that is unsupported on this system, attempting to build anyway.
Info: Internal Builder is used for build
gcc -O0 -g3 -Wall -c -fmessage-length=0 -o "src\\hello.o" "..\\src\\hello.c" 
Cannot run program "gcc": Launching failed

I can't find any clear documentation to resolve this.  I'm running windows 10 pro

**Comment by jantje** [comment]
What does your OS say when you run the command from the command prompt?

**Comment by rsail** [comment]
What command should I run in the os?

**Comment by jantje** [comment]
The command CDT wants to run and returns with an error. In this case that would be
`gcc -O0 -g3 -Wall -c -fmessage-length=0 -o "src\hello.o" "..\src\hello.c"`

**Reply by ghost** [reply]
When I run gcc -O0 -g3 -Wall -c -fmessage-length=0 -o "src\hello.o" "..\src\hello.c"  I get:

'gcc' is not recognized as an internal or external command,
operable program or batch file.

But when I run MSYS UCRT64 and run gcc --version I get

gcc --version
gcc.exe (Rev7, Built by MSYS2 project) 13.1.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

My Path variable is set to PATH=C:\msys64\mingw64\bin;.....

but I notice there is nothing in that folder which seems weird

**Reply by jld01** [reply]
The [mingw-w64-x86_64-toolchain](https://packages.msys2.org/groups/mingw-w64-x86_64-toolchain) is installed under the `mingw64` folder.
The [mingw-w64-ucrt-x86_64-toolchain](https://packages.msys2.org/groups/mingw-w64-ucrt-x86_64-toolchain) is installed under the `ucrt64` folder.

**Reply by ghost** [reply]
can someone please provide complete step by step instructions on how to install a c compiler to work with eclipse?  Why the heck is this so hard to find clear information on how to do this.

**Reply by ghost** [reply]
It seems now I getting unresolved inclusion for all my *.h files and EXIT_SUCCESS could not be resolved.  here's the actual program:

```
/*
 ============================================================================
 Name        : hello.c
 Author      : RS
 Version     :
 Copyright   : Your copyright notice
 Description : Hello World in C, Ansi-style
 ============================================================================
 */

#include <stdio.h>
#include <stdlib.h>

int main(void) {
	printf("!!!Hello World!!!"); /* prints !!!Hello World!!! */
	return EXIT_SUCCESS;
}

```

**Reply by ghost** [reply]
ok making some progress,  I have the following screenshot:

![image](https://github.com/eclipse-cdt/cdt/assets/6682434/cfe51b85-d2f9-4698-8f37-ae1d96cefd8a)


I have the MinGW added to the preprocessor (treat as built in, contains system headers). EXIT_SUCCESS looks resolved now but I still can't resolve my h files.

**Reply by ghost** [reply]
EXIT_SUCCESS still not resolved! wow, this is really painful
